### DISCUSSION TITLE
The dense links in demo are very scary

### DISCUSSION THREAD

**Discussion by xjArea** [vikejs/vike#991] [Discussion] [answered]
😊 But the first time I executed `npm run dev`, I was very happy to see the project successfully launched , 😨 but later I found that the network panel had up to **84** links, which scared me. 😖 Demos are so complex, and development must be difficult. 😕 Later, I found that most of the links were caused by scattered imports, and these links would disappear after build.

😳 Nevertheless, I still believe that the **84** links in the development phase are too many, which greatly interferes with checking links during development. Therefore, I am wondering if there are other ways besides build to reduce the number of links and make the project look more refreshing and concise.

😓 Here are the relevant screenshots. As a demo, the loading time is also quite long

![image](https://github.com/brillout/vite-plugin-ssr/assets/35623539/be863e65-d7a6-4c86-acb8-42a8adfec3d7)

**Comment by brillout** [comment, marked as answer]
That's how Vite works. (And that's a good thing.)

**Reply by xjArea** [reply]
😲 I don't really understand "That's how Vite works.", I executed `npm run dev` on the Vite project, and found that it doesn't have so many links, and it is much more concise than Vite-Plugin-SSR base demo.

😅 Now I use `-node_modules` to filter the network so that it does not display the content of node_module. I really hope there is a way to not display so many links, because it interferes with project development, Is there really no other way to improve it?

**Reply by brillout** [reply]
Webpack compiles all files into a single bundle, and that's a reason why webpack is so slow. Vite doesn't do that in development and instead transpiles on-demand only what's needed, which makes Vite *much* faster. So it's a good thing.

>  it interferes with project development

How so?

**Reply by xjArea** [reply]
I know that webpack will package files, but if you look carefully at those links, they are all caused by Vite-Plugin-SSR using `import`. Vite’s basic example does not have so many links.

Too many links are not good for us to check the project, because it will cause interference, and it will affect our judgment on the time speed of the link and the size of the file during development. This is a matter of course, so I will look for a solution.

Of course, it doesn’t matter if there is no way to solve it. I can use the keyword `-node_modules` to filter, but as a demo, the simpler the better, isn’t it? 😁

**Reply by brillout** [reply]
Vite's way is indeed unconventional, but it's fundamentally superior to the old webpack way.

At the end of the day, what matters here is dev speed. Compare Vite and alternatives and see for yourself 😉. 

(In case you're curious: the [V1 design](https://vite-plugin-ssr.com/migration/v1-design) slightly reduces the number of `import()` statements to the most minimal amount possible).

**Reply by xjArea** [reply]
Oh, reduces `import()`, that is a good news, hope the project will be better in the future, yeah. 😄
